sdk/core/Azure.Core/src/ResourceType.cs

```diff
@@ -144,5 +131,10 @@ public override int GetHashCode()
         {
             return StringComparer.OrdinalIgnoreCase.GetHashCode(_stringValue);
         }
+
+        internal ResourceType AppendChild(string child)
```

Comment:
`AppendChild` is always called for a `ReadOnlySpan` that is converted to string. Do we care about optimizing it?

Reply:
Will do